extensions/stats/plugin.cc

```diff
@@ -561,17 +578,32 @@ bool PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
     uint64_t response_flags = 0;
     getValue({"response", "flags"}, &response_flags);
     if (peer_node_info.maybeWaiting() && response_flags == 0) {
-      return false;
-    }
-    if (!request_info.is_populated) {
-      ::Wasm::Common::populateTCPRequestInfo(outbound_, &request_info);
+      return;
     }
+    ::Wasm::Common::populateTCPRequestInfo(outbound_, &request_info);
   } else {
-    ::Wasm::Common::populateHTTPRequestInfo(outbound_, useHostHeaderFallback(),
-                                            &request_info);
+    // Populate HTTP request info fully only at the end of the stream because
+    // onTick context has no access to request/response headers but can read
+    // from filter state.
+    if (end_stream) {
+      ::Wasm::Common::populateHTTPRequestInfo(
+          outbound_, useHostHeaderFallback(), &request_info);
+    } else {
+      // Await for upstream host selection to resolve destination service.
+      int64_t port;
+      if (!getValue({"upstream", "port"}, &port)) {
+        return;
+      }
+      ::Wasm::Common::populateRequestInfo(outbound_, useHostHeaderFallback(),
+                                          &request_info);
+      if (request_info.request_protocol == Protocol::GRPC) {
```

Comment:
curious, can report be called for HTTP onTick?

Reply:
Yes, see the test. It's called for gRPC streams only right now.